internal/db/fetcher/encoded_doc.go

```diff
@@ -68,8 +68,8 @@ type encodedDocument struct {
 	id                   []byte
 	schemaVersionID      string
 	status               client.DocumentStatus
-	properties           map[client.FieldDefinition]*encProperty
-	decodedPropertyCache map[client.FieldDefinition]any
+	properties           map[client.FieldDefinitionKey]*encProperty
```

Comment:
question: Why are you using both field ID and Name here?  At a single point in time either will be unique - you can't two fields with the same fieldID or with the same name.

Reply:
Both the field ID and name are used in the downstream dependencies. If we used only one as the map key then we would need to expand the `FieldDefinition` API to support looking up a field by ID or copy paste a lookup function into a few places. I think having the `FieldDefinitionKey` is the better alternative.

Reply:
There should already be functions to get fields by `Name` if you use that as the key - I don't think you'd need to add anything to `client`.

I see having a (public) `FieldDefinitionKey` struct with `name` and `key` properties suggests that they need to be combined in order to form a unique key - which is incorrect.

Reply:
The `DecodeToDoc` function in the fetcher doesn't have a reference to the definition so it can't lookup the field ID from the name.

I'm fine with changing the struct name but I don't see any issues with the way I implemented it.

Reply:
`DecodeToDoc` just uses the field ID as an index - we could change that too.  Or you could change `encodedDocument`.`properties` to a slice containing field name and value.  

There are many ways to solve this internal coding problem that don't involve changing the public types, especially not in a way that seems a little misleading.